docs(agents): update with v1.0 rcs out (#5071)
dapr/docs: daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-core-concepts.md

| `ConversationVectorMemory` | Vector Store | ✅ | Semantic | RAG/AI Apps |
| `ConversationDaprStateMemory` | Dapr State Store | ✅ | Query | Production |

`ConversationVectorMemory` can be backed by any of the supported vector store implementations:

| Vector Store | Class | Backend | Notes |
|---|---|---|---|
| Chroma | `ChromaVectorStore` | ChromaDB | In-memory or persistent; no extra infrastructure |
| PostgreSQL | `PostgresVectorStore` | pgvector extension | Requires PostgreSQL with `pgvector` |
| Redis | `RedisVectorStore` | Redis Stack / Redis with Search | Requires `redisvl` |

```python
from dapr_agents.storage.vectorstores import RedisVectorStore
from dapr_agents.document.embedder.openai import OpenAIEmbedder
from dapr_agents.memory import ConversationVectorMemory

vector_store = RedisVectorStore(
    url="redis://localhost:6379",
    index_name="my_agent",
    embedding_function=OpenAIEmbedder(),
    embedding_dimensions=1536,
)

memory = ConversationVectorMemory(
    vector_store=vector_store,
    distance_metric="cosine",
)
```

### Agents as Tools

> Dapr Agents supports invoking other agents—whether Dapr Agents or agents from other frameworks—as tools within a `DurableAgent`'s reasoning loop. This lets a parent agent delegate sub-tasks to specialized child agents and compose multi-agent systems without event-driven pub/sub.

Reviewer: ```suggestion
Dapr Agents supports invoking other agents - whether Dapr Agents or 3rd party agent frameworks - as tools within a `DurableAgent` reasoning loop. This lets a parent agent delegate sub-tasks to specialized child agents and compose multi-agent systems without using a pub/sub message broker.
```

> Agents registered in the same registry are available as tools automatically. This includes invoking other framework agents. Alternatively, use `agent_to_tool` from `dapr_agents.tool.workflow` for explicit wiring, cross-app routing, or invoking agents from other frameworks:

Reviewer: ```suggestion
Agents registered in the same registry are available to use as tools automatically. This includes invoking 3rd party framework agents. Alternatively, use `agent_to_tool` from `dapr_agents.tool.workflow` for explicit wiring, cross-app routing, or invoking agents from other frameworks:
```

```python
from dapr_agents.tool.workflow import agent_to_tool

> # Different Dapr app (cross-app)

Reviewer: ```suggestion
# Invoke a separate agent as a tool call
```

aragorn_tool = agent_to_tool(
    "aragorn",
    description="Military Strategy. Goal: Lead the forces of Gondor.",
    target_app_id="aragorn-app",
)

> 

Reviewer: ```suggestion
# Use an agent as a tool within a DurableAgent
```

frodo = DurableAgent(
    name="frodo",
    role="Ring Bearer",
    goal="Carry the One Ring to Mordor",
    tools=[aragorn_tool],
    ...
)
```

> When the LLM calls one of these tools, Dapr Agents schedules the target agent's workflow as a child workflow and returns the result—handling cross-app routing and result marshalling transparently.

Reviewer: ```suggestion
When the LLM calls one of these tools, Dapr Agents schedules the target agent's workflow as a `DurableAgent` (child workflow) and returns the result—handling cross-app routing and result marshalling transparently.
```

Reviewer: We seem to be expecting people to have knowledge of cross-app and Dapr workflow concepts in these docs which i dont necessarily think should be the case.
